### PR TITLE
feat: add mockFunction and other testFunctions

### PR DESCRIPTION
**Pull request by bchelkowski**
BREAKING CHANGE: adds insertKopytkoUnitTestSuiteArgument bs_const flag

## What did you implement:

Implemented:
- test function can now return an array of asserts
- mockFunction - operates on the test suite `__mocks` object
- it - shorthand for ts.addTest
- itEach - shorthand for ts.addParameterizedTests
- beforeAll - shorthand for ts.setBeforeAll
- beforeEach - shorthand for ts.setBeforeEach
- afterEach - shorthand for ts.setAfterEach
- afterAll - shorthand for ts.setAfterAll

Added documentation

## How did you implement it:

Added new functions.

## How can we verify it:

Run unit tests.
There is one test suite where all test cases are supposed to fail.

Tested on Roku DAZN app, to verify if changes break anything - everything seems to work.

## Todos:

- [x] Write documentation (if required)
- [x] Fix linting errors
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** YES

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-- [App Structure](#app-structure)
-- [Setup](#setup)
-- [Running Unit Tests](#running-unit-tests)
-- [Kopytko Unit Test Philosophy](#kopytko-unit-test-philosophy)
-- [Test Mocks](#test-mocks)
-- [Setup and Teardown](#setup-and-teardown)
-- [Limitations](#limitations)
-- [API](#api)
-- [Example test app config and unit tests](#example-test-app-config-and-unit-tests)
+- [Kopytko Unit Testing Framework](#kopytko-unit-testing-framework)
```
Comment (pawelhertman) [inline README.md:3]: Why does it need this top-level entry?

File: README.md [hunk 2]
```diff
@@ -111,9 +112,9 @@ end function
 function MyServiceTestSuite() as Object
   ts = KopytkoTestSuite()
 
-  ts.setBeforeAll = sub (ts as Object)
+  beforeAll(sub (_ts as Object)
```
Comment (pawelhertman) [inline README.md:115]: I know it was overwriting MyServiceTestSuite's ts variable but shall we care about it?

File: README.md [hunk 3]
```diff
@@ -215,14 +216,14 @@ The service can be used like a regular object:
 
 Calls to the methods or constructor can be inspected:
 ```brightscript
-?m.__mocks.exampleService.getData.calls[0].params.arg
-?m.__mocks.exampleService.constructorCalls[0].params.dependency
+? mockFunction("ExampleService.getData").getCalls()[0].params.arg
```
Comment (pawelhertman) [inline README.md:219]: could you be a hero and add a simple function mock example?

**Review by pawelhertman — COMMENTED** [review]
Awesome job 👏 I'm just not sure about `_ts`

File: docs/api/KopytkoExpect.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+  - [Methods](#methods)
+    - [`toBe`](#tobe)
+    - [`toBeInvalid`](#tobeinvalid)
+    - [`toBeValid`](#tobevalid)
+    - [`toBeTrue`](#tobetrue)
+    - [`toBeFalse`](#tobefalse)
+    - [`toEqual`](#toequal)
+    - [`toContain`](#tocontain)
+    - [`toHasKey`](#tohaskey)
+    - [`toHasKeys`](#tohaskeys)
```
Comment (pawelhertman) [inline docs/api/KopytkoExpect.md:17]: wouldn't it be better to call it `toHaveKey(s)`?

Comment (bchelkowski) [reply docs/api/KopytkoExpect.md:17]: Yes, it would

File: README.md (hunk 2 above)
Comment (bchelkowski) [reply README.md:115]: Don't really know what you mean.

File: README.md (hunk 1 above)
Comment (bchelkowski) [reply README.md:3]: My markdown extension did it automatically.
I turned off it now, should be fixed

File: README.md (hunk 3 above)
Comment (bchelkowski) [reply README.md:219]: However, there are plenty examples in our Example app, but I will add here something

**Comment by bchelkowski** [comment]
About this `_ts`...
I would be careful about removing it now. Because I'm not sure if rewriting all our test cases at once would be a good option.
What I can offer is adding a bs_const something like `kopytkoSimpleUnitTest: true` to our manifest, it would be temporary.
If this will be `true`, you won't need to have this argument.
If `false`, it will be in the old way.

**Review by pawelhertman — APPROVED** [review]

**Review by kuldeepk-dazn — APPROVED** [review]

**Pull request merged**